service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java

```diff
@@ -157,6 +158,23 @@ void constructDirectionForCCD_shouldConstructDirectionFromConfigurationAsExpecte
             .build());
     }
 
+    @Test
+    void numberDirections_shouldApplyCorrectNumberingWhenDirectionsAreMarkedAsRemoved() {
+        CaseData caseData = populateCaseDataWithSomeDirectionsMarkedAsRemoved().build();
+
+        List<Element<Direction>> directions = service.combineAllDirections(caseData);
+
+        List<String> numberedDirectionTypes = service.numberDirections(directions).stream()
+            .map(direction -> direction.getValue().getDirectionType())
+            .collect(toList());
+
+        List<String> expectedDirectionsTypes = IntStream.range(0, numberedDirectionTypes.size())
```

Comment:
would it be worth renaming 'x' to something meaningful here like directionNumber?

Reply:
updated